app/models/ability.rb

```diff
@@ -245,6 +245,8 @@ def package_abilities
         :undesignate_partial_item, :dispatch_stockit_item, :move_stockit_item,
         :move_partial_quantity, :move_full_quantity, :print_inventory_label,
         :undispatch_stockit_item, :stockit_item_details, :split_package], Package
+    elsif can_search_browse_packages?
+      can [:index, :show], Package, { allow_web_publish:true }
```

Comment:
Yeah @steveyken , I though of that too, actually whenever user is logged in, it first goes to the `public_ability` method and then the `package_ability` method. 

In `package_ability` method, we have `can_manage_packages?` condition if it turns out to be false(which will be false for the charity users), the code runs to else block, where we are getting the donor packages with current user. 
That causes to return `nil` and there are no packages shown on the browse app. So that's the main reason for this permission which i have added for charity_user. 

cc- @namrataukirde

Reply:
ah good catch! One of the problems with blocks in CanCanCan is that it doesn't merge well with the other scopes.

I think if we change that block from

```
 can [:index, :show, :create, :update], Package, Package.donor_packages(@user_id) do |record|
        record.item ? record.item.offer.created_by_id == @user_id : false
      end
```

to 

```can [:index, :show, :create, :update], Package, item: { offer: {created_by_id: @user_id }```

then it should merge correctly with the ability defined in public_ability. We could then remove `Package.donor_packages` completely

Reply:
Yeah, this makes sense, Will try it out on Monday, and send the PR for re-review.
Thanks @steveyken